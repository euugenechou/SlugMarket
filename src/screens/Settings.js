/* React imports */
import React, { Component } from 'react';
import { Alert, Platform, Text, View, StyleSheet, Button, SafeAreaView, Image } from 'react-native';
import LoginStack from './login/LoginStack'
import SettingsList from 'react-native-settings-list';
import {API, Auth} from "aws-amplify"
import UserListings from "./components/Profile/UserListings";


export default class Settings extends Component {
	static navigationOptions = ({ navigation }) => ({
		header: null
	})

	constructor(props){
		super(props);
		this.onValueChange = this.onValueChange.bind(this);
		this.state = {switchValue: false};
		this.state = {
			activeIndex: 0,
			postsToRender: [],
			userName: "",
			userAttributes: {},
			refreshing: false,
			switchValue: false,
		  };
	  }

	getUserInfo() {
		Auth.currentUserInfo()
		  .then(res => {
			this.setState({ userAttributes: res.attributes });
			this.setState({ userName: res.username });
			console.log(res);
		  })
		  .catch(err => console.log(err));
	}
	
	componentWillMount() {
	this.startHeaderHeight = 60;
	this.getUserInfo();
	console.log(this.state.postsToRender);
	}
	
	 _onPressButton() {
	 	() => this.props.navigation.navigate("SignInScreen");
	// 	// Auth.signOut()
	// 	// 	.then(() => {
				
	// 	// 	})
	// 	// 	.catch(err => console.log(err));
	}

	onValueChange(value){
		this.setState({switchValue: value});
	  }

	render() {
		return (
		<View style = {{backgroundColor:'white',flex:1}}>
			<View style = {{flex:1, marginTop:25}}>
				<SettingsList>
					<SettingsList.Header headerText='Settings' headerStyle={{color:'teal', fontWeight: 'bold', fontSize: 24}}/>
					{/* <SettingsList.Item
						icon = {
							<View style = {{height:100, marginLeft:10, alignSelf:'center'}}>
								<Image style = {{alignSelf:'center', height:100, width:100}} source={require("../assets/darrell.png")}/>
							</View>
						}
						itemWidth = {50}
						hasNavArrow ={ false}
						fontSize = {50}
						title = {this.state.userName}
					/> */}
					<SettingsList.Header headerText = 'Location' headerStyle={{color:'white', marginTop:50}}/>
					<SettingsList.Item	
						title = 'Edit location'
						//hasNavArrow = {true}
						arrowStyle = {{tintColor:'teal'}}
						//onPress = {() => {Alert.alert('Change location')}}/>
						/>
					<SettingsList.Header headerText = 'Authors' headerStyle={{color:'white', marginTop:50}}/>
						<SettingsList.Item
								title='Authors'
								hasNavArrow={true}
								arrowStyle={{tintColor:'blue'}}
								onPress={() =>
									this.props.navigation.navigate("AuthorsScreen")
								}
							/> 
					</SettingsList>
				<View>
					<Button
						raised
						color = "teal"
						backgroundColor = "white"
						borderRadius = {2}
						title = "Sign Out"
						fontWeight = "bold"
						onPress={() =>
							this.props.navigation.navigate("SignInScreen")
						}
						containerViewStyle={{ width: 150, paddingTop: 10 }}
					/>
				</View>
			</View>
		</View>
    );
  }
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		alignItems: 'center',
		justifyContent: 'center',
	},
	bottomView:{
		flex: 1,
		width: '100%', 
		height: 50, 
		backgroundColor: '#FF9800', 
		justifyContent: 'center', 
		alignItems: 'center',
		position: 'absolute',
		bottom: 0
	},
	settings:{
		flex: 1,
		fontSize: 30, 
		justifyContent: 'flex-start',
	  },
	  row: {
		flexDirection: "column",
		justifyContent: "center",
		padding: 10,
		backgroundColor: "#f6f6f6",
		marginBottom: 8
	  },
	  text: {
		flex: 1
	  }
});