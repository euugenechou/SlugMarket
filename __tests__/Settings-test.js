// __tests__/Settings-test.js
import 'react-native';
import React from 'react';
import Settings from '../src/screens/Settings';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Settings />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});