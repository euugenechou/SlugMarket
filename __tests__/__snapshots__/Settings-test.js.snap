// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "paddingHorizontal": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginTop": 25,
      }
    }
  >
    <RCTScrollView>
      <View>
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "margin": 5,
                },
                Object {
                  "color": "black",
                  "fontSize": 36,
                  "fontWeight": "700",
                  "paddingLeft": 10,
                },
              ]
            }
          >
            Settings
          </Text>
          <View
            style={
              Object {
                "borderBottomWidth": 1,
                "borderColor": "white",
                "borderTopWidth": 1,
              }
            }
          >
            <View
              accessible={false}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    Object {
                      "backgroundColor": "white",
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "flexDirection": "row",
                        "marginLeft": 15,
                      },
                      Object {
                        "borderBottomWidth": 1,
                        "borderColor": "white",
                      },
                      Object {
                        "minHeight": 70,
                      },
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontSize": 18,
                        },
                        Object {
                          "alignSelf": "center",
                          "flex": 1,
                        },
                      ]
                    }
                  >
                    View/Edit Profile
                  </Text>
                  <Image
                    source={
                      Object {
                        "testUri": "../../react-native-settings-list/img/icon-arrow-settings.png",
                      }
                    }
                    style={
                      Array [
                        Object {
                          "alignSelf": "center",
                          "marginRight": 15,
                        },
                        Object {
                          "tintColor": "teal",
                        },
                      ]
                    }
                  />
                </View>
              </View>
            </View>
            <View
              accessible={false}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    Object {
                      "backgroundColor": "white",
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "flexDirection": "row",
                        "marginLeft": 15,
                      },
                      Object {
                        "borderBottomWidth": 1,
                        "borderColor": "white",
                      },
                      Object {
                        "minHeight": 70,
                      },
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontSize": 18,
                        },
                        Object {
                          "alignSelf": "center",
                          "flex": 1,
                        },
                      ]
                    }
                  >
                    Terms of Service
                  </Text>
                  <Image
                    source={
                      Object {
                        "testUri": "../../react-native-settings-list/img/icon-arrow-settings.png",
                      }
                    }
                    style={
                      Array [
                        Object {
                          "alignSelf": "center",
                          "marginRight": 15,
                        },
                        Object {
                          "tintColor": "teal",
                        },
                      ]
                    }
                  />
                </View>
              </View>
            </View>
            <View
              accessible={false}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    Object {
                      "backgroundColor": "white",
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "flexDirection": "row",
                        "marginLeft": 15,
                      },
                      Object {
                        "borderWidth": 0,
                      },
                      Object {
                        "minHeight": 70,
                      },
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontSize": 18,
                        },
                        Object {
                          "alignSelf": "center",
                          "flex": 1,
                        },
                      ]
                    }
                  >
                    SlugMarket v0.1.0
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View>
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            Object {
              "shadowColor": "rgba(0,0,0, .4)",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 1,
              "shadowRadius": 1,
            },
            Object {
              "alignSelf": "center",
              "paddingBottom": 40,
              "width": 300,
            },
            Object {
              "borderRadius": 5,
            },
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "red",
                },
                Object {
                  "borderRadius": 5,
                },
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    },
                    Object {
                      "color": "white",
                    },
                    Object {
                      "fontSize": 17.5,
                    },
                    undefined,
                    undefined,
                    Object {
                      "fontWeight": "bold",
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Sign Out
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
